docker/docker_client.go

```diff
@@ -502,7 +502,10 @@ func (c *dockerClient) makeRequestToResolvedURL(ctx context.Context, method, url
 			attempts == backoffNumIterations {
 			return res, err
 		}
-
+		// close response body befor retry or context done
+		if res != nil {
```

Comment:
`make validate` isn't happy with this condition.

Reply:
remove the if statement